src/math/fft_base.cpp

```diff
@@ -0,0 +1,47 @@
+#include "fft_base.h"
+
+namespace FFT
+{
+    BaseFFTExecutor::BaseFFTExecutor() 
+        : _width(0)
+        , _height(0)
+    {
+    }
+
+    BaseFFTExecutor::BaseFFTExecutor( uint32_t width_, uint32_t height_ )
+        : _width( width_ )
+        , _height( height_) 
+    {
+        // Need to call initialize() from sub-class as intialize calls virtual functions.
+    }
+
+    void BaseFFTExecutor::initialize( uint32_t width_, uint32_t height_ )
+    {
+        if( width_ == 0 || height_ == 0 )
+            throw imageException( "Invalid parameters for FFTExecutor::intialize()" );
+
+        _clean();
+
+        _makePlans(width_, height_);
```

Comment:
Move this function call at the end of the function and remove input parameters as you can use class members.

Reply:
Done.